quic/s2n-quic-transport/src/endpoint/mod.rs

```diff
@@ -103,26 +103,18 @@ impl<Cfg: Config> s2n_quic_core::endpoint::Endpoint for Endpoint<Cfg> {
 
         let local_address = queue.local_address();
         let entries = queue.as_slice_mut();
-        let mut now: Option<Timestamp> = None;
+        let timestamp = clock.get_time();
```

Comment:
I assumed not, the entry.read is not async and it shouldnt take very long since its just seems to be returning a header info and an index into a buffer (the receiving has been done from what i can tell).

Reply:
The idea was to not query the clock unless we absolutely need to. Your change now queries it regardless if those lists are empty.

I think we can make an abstraction, though, to make this easier to consume.

Reply:
reverted the code to maintain the previous behavior.